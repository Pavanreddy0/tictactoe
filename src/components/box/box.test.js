import { render, cleanup } from "@testing-library/react";
import Box from "./box";
import React from "react";
import ReactDOM from "react-dom";
import renderer from "react-test-renderer";

afterEach(cleanup);

it("button renders without crashing", () => {
  const div = document.createElement("div");
  ReactDOM.render(<Box />, div);
});

it("button renders successfully", () => {
  const { getByTestId } = render(<Box value="Undo" />);
  expect(getByTestId("box")).toHaveTextContent("Undo");
});

it("matches snapshow", () => {
  const tree = renderer.create(<Box value="Refresh"></Box>).toJSON();
  expect(tree).toMatchSnapshot();
});

it("matches snapshot 2", () => {
  const tree = renderer.create(<Box value="Pavan"></Box>).toJSON();
  expect(tree).toMatchSnapshot();
});
