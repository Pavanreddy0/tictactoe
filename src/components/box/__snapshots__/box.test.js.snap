// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 2 1`] = `
<button
  data-testid="box"
  style={
    Object {
      "cursor": "pointer",
      "fontSize": "40px",
      "styleborder": "3px solid black",
    }
  }
>
  Pavan
</button>
`;

exports[`matches snapshow 1`] = `
<button
  data-testid="box"
  style={
    Object {
      "cursor": "pointer",
      "fontSize": "40px",
      "styleborder": "3px solid black",
    }
  }
>
  Refresh
</button>
`;
